### PR TITLE
[feature] Clear fixture directory before generating new golden files

### PR DESCRIPTION
**Pull request by ltpquang**
Hi all, this PR intends to improve issue #19.

Sometimes if we update test cases that affect the golden files' names and then run `go test -update`, the old named files are left untouched.

This PR presents `-clean` flag, which will clear and recreate output directory of golden files.

### CODE REVIEW

**Comment by sebdah** [comment]
Thank you for this PR. I've been reading this through but do not really understand the need of the logic based on the timestamp. I would have assumed a process where we are:

(if `-clean` is provided)
1. Remove all files in the testdata directory
2. Update all testdata files

But maybe I'm missing some case here.

**Comment by ltpquang** [comment]
@sebdah 

### TL;DR
If two (or more) test cases produces golden files to the same directory, the former will clean the output dir and create golden files, but the later will also do the same, making the formers' golden files removed.

### Example
Say my `worker_test.go` have two table test cases:

```
func TestCaseA(t *testing.T) {
	tests := []struct {
		name string
		input int
	}{
		{
			name: "CaseB",
			input: 1,
		},
	}
	gd := New(t)
	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			myWorker := NewWorker()
			workingResult := myWorker.DoSomething(test.input)
			gd.Assert(t, test.name, workingResult)
		})
	}
}

func TestCaseB(t *testing.T) {
	tests := []struct {
		name string
		input string
	}{
		{
			name: "CaseB",
			input: "input string",
		},
	}
	gd := New(t)
	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			myAnotherWorker := NewAnotherWorker()
			workingResult := myAnotherWorker.DoSomethingElse(test.input)
			gd.Assert(t, test.name, workingResult)
		})
	}
}
```

The `TestCaseA`, and `TestCaseB` are supposed to create golden files `CaseA`, and `CaseB`. Since I initiated goldies with only one compulsory param, no naming or directories stuff, both test cases will produce golden files to the same default directory `testdata`.

If we use `-update -clear`, the `TestCaseA` will clear the directory before creating `CaseA` file. But right after that the `TestCaseB` will do the same also, which makes the `CaseA` disappear.

Therefore I use a timestamp as a simple lock. This timestamp is initiated once when the `go test` is executed. `goldie` then will try to update the golden files' directory Modified Time to this timestamp, everytime a golden file is created.

So that the very first `Assert` call, when checking for the output directory, will know that it is the first one to come to the dir (current timestamp is different than the dir timestamp, no one has came before), and its job is to clean the dir first. The subsequence `Assert` call will not try to clean the dir anymore, because the current timestamp is the same as the dir's Modified Time

**Comment by sebdah** [comment]
Understood. Thank you, I'm merging this now :rocket:
